scheduler/core/programprovider/ocs/ocsprogramprovider.py

```diff
@@ -79,6 +79,18 @@ def read_ocs_zipfile(zip_file: str | PathLike[str], program_ids: Optional[Frozen
                 logger.info(f'Skipping program {program_id} as it is not in the list.')
 
 
+def parse_preimaging(sequence: List[dict]) -> bool:
+
+    preimaging = False
+    try:
+        if sequence[0][OcsProgramProvider._AtomKeys.PREIMAGING].upper() == 'YES':
+            preimaging = True
+    except:
```

Comment:
For future reference, bare exception clauses are highly recommended against.
This should probably be `KeyError`, since I think that is likely the only exception that could come up in this circumstance (unless the `sequence` might be empty, in which case, also `IndexError`.